### ISSUE TITLE
random.choice gives results even with spurious values for 'replace'

### ISSUE THREAD

**Issue by adivijaykumar** [numpy/numpy#12110]
random.choice does not raise an error when `replace` keyword is set to anything other than `True` or `False`. In general, setting the  `replace` keyword  to any string or float does not give an error.

Is this how it is supposed to be, or is this a bug? If it isn't, I think it would at least be worthy including in the documentation for `replace`.

### Reproducing code example:

```python
import numpy as np
print np.random.choice([1,2,3,4,5],3,replace=2), np.random.choice([1,2,3,4,5],3,replace='asfasdlol')
```

### Error message:
(Output)
[2 1 1] [3 3 1]

### Numpy/Python version information:
('1.15.2', '2.7.15 |Anaconda custom (64-bit)| (default, May  1 2018, 23:32:55) \n[GCC 7.2.0]')

**Comment by seberg** [comment]
I don't think this is a bug, it would be probably be a bigger discussion if we would want to do input sanitation in such cases.
Basically, all or at least most such boolean flags simply use `bool(replace)` logic. I would like to see an example or counter example in the python standard lib, but I could not find it.
However, changing it could be a bit tricky anyway, since `True is not np.bool_(True)` and so on.

**Comment by adivijaykumar** [comment]
I was referring to input sanitation when I said 'bug'. Sorry for that.

I am not aware of the specifics, but had reported this because the behaviour seemed to be unlike any other method where a a boolean flag is used (they report some error).

**Comment by mattip** [comment]
Closing as known behaviour
